acme.go

```diff
@@ -116,7 +117,8 @@ type AuthzID struct {
 // Client implements ACME spec.
 type Client struct {
 	http.Client
-	Key *rsa.PrivateKey
+	// Key.Public() must be *rsa.PublicKey
+	Key crypto.Signer
```

Comment:
I don't think separating the two is a good idea, tbqh. And this is the groundwork for an upcoming PR for ECDSA support, so having it be *rsa.PublicKey explicitly doesn't help.